dali/pipeline/pipeline.cc

```diff
@@ -235,6 +236,43 @@ void Pipeline::AddOperator(OpSpec spec, const std::string& inst_name) {
   this->op_specs_to_serialize_.push_back(true);
 }
 
+inline int GetMemoryHint(OpSpec &spec, int index) {
+  std::vector<int> hints;
+  GetSingleOrRepeatedArg(spec, &hints, "bytes_per_sample_hint", spec.NumOutput());
+
+  if (index < (int)hints.size())
+    return hints[index];
+  else
+    return 0;
```

Comment:
If index is > than hints.size(), this means that index is > spec.NumOutput() and we should fail. Maybe we should use DALI_ENFORCE for that.

Reply:
User might have specified hint just for one output. We should not fail, it's totally optional. On the other hand, we might fail if there are more hints than outputs.

Reply:
This will be handled in the GetSingleOrRepeatedArg. If he provide only one hint, then GetSingleOrRepeatedArg will duplicate it for all outputs. If there is more hints or less than num of outputs but moire than one it will fail to there.